### ISSUE TITLE
Early stopping 

### ISSUE THREAD

**Issue by marjanfamili** [alan-turing-institute/affinity-vae#25]
Implement early stopping (early stopping of the training) if the log of the loss has not changed for threshhold number of epoch

This should be optional as we also will have the ability to save the state of the model and evaluate by loading the desired state

**Comment by marjanfamili** [comment]
Early_stopping_meta
Early_stropping_state

**Comment by crangelsmith** [comment]
We have state files, f1 scores and reconstruction figures every N epochs (N is controlled by the freq_all flag). If early stopping was needed we can always recover a previous state from the saved files.

An attempt for early stopping was made in PR #219, but there is too many different reasons for early stopping (e.g overfitting on one of the 3 losses, degradation of reconstruction, loss of accuracy) but it is always a trade-off that has to be done depending of what is the goal of the training so I've decided to not add the implementation as it just makes the code larger and we already can recover early states.

I'll close this for now, but the early stopping branch is there.
